### PR TITLE
add a -c option to the benchmarks to print the cmake configuration

### PR DESCRIPTION
**Pull request by angainor**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,6 +75,10 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/config.hpp.in
     ${CMAKE_CURRENT_BINARY_DIR}/include/oomph/config.hpp @ONLY)
 install(FILES ${PROJECT_BINARY_DIR}/include/oomph/config.hpp
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/oomph)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_config.hpp.in 
+    ${CMAKE_CURRENT_BINARY_DIR}/include/oomph/cmake_config.hpp)
```
Comment (boeschf) [inline CMakeLists.txt:79]: can we rename the resulting ending in .inc: `cmake_config.inc` in order to make it clear, that this is a code snippet?

File: benchmarks/args.hpp [hunk 2]
```diff
@@ -27,7 +29,12 @@ struct args
 
     args(int argc, char** argv)
     {
-        if (argc != 4) { is_valid = false; }
+        if (argc != 4) { 
+	    is_valid = false; 
+	    if(argc==2 && !std::strcmp(argv[1], "-c")){
+	        print_config(argv[0]);
+	    }
+	}
```
Comment (boeschf) [inline benchmarks/args.hpp:37]: can you clang-format your changes?

File: cmake/cmake_config.hpp.in [hunk 3]
```diff
@@ -0,0 +1,15 @@
+std::cout << "CMAKE_CXX_FLAGS=" << "@CMAKE_CXX_FLAGS@" << std::endl;
+std::cout << "CMAKE_C_FLAGS=" << "@CMAKE_C_FLAGS@" << std::endl;
+std::cout << "CMAKE_BUILD_TYPE=" << "@CMAKE_BUILD_TYPE@" << std::endl;
+
+std::cout << "OOMPH_UCX_USE_PMI=" << "@OOMPH_UCX_USE_PMI@" << std::endl;
+std::cout << "OOMPH_UCX_USE_SPIN_LOCK=" << "@OOMPH_UCX_USE_SPIN_LOCK@" << std::endl;
+std::cout << "OOMPH_USE_FAST_PIMPL=" << "@OOMPH_USE_FAST_PIMPL@" << std::endl;
+std::cout << "OOMPH_UCX_USE_MULTIPLE_ENDPOINTS=" << "@OOMPH_UCX_USE_MULTIPLE_ENDPOINTS@" << std::endl;
```
Comment (boeschf) [inline cmake/cmake_config.hpp.in:8]: can we leave this one out until we have this option?

File: CMakeLists.txt (hunk 1 above)
Comment (angainor) [reply CMakeLists.txt:79]: done

File: cmake/cmake_config.hpp.in (hunk 3 above)
Comment (angainor) [reply cmake/cmake_config.hpp.in:8]: oh, sure.

File: benchmarks/args.hpp (hunk 2 above)
Comment (angainor) [reply benchmarks/args.hpp]: hope it's better now. manual work..

**Pull request closed without merging**